Assets/MRTK/Providers/WindowsMixedReality/XR2018/Controllers/HPMotionController.cs

```diff
@@ -0,0 +1,271 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using Unity.Profiling;
+using UnityEngine;
+using UnityEngine.XR;
+using Microsoft.MixedReality.Input;
+using System;
+
+namespace Microsoft.MixedReality.Toolkit.WindowsMixedReality.Input
+{
+    public class MotionControllerState
+    {
+        public MotionControllerState(MotionController mc)
+        {
+            this.MotionController = mc;
+        }
+        public void Update(DateTime currentTime)
+        {
+            this.CurrentReading = MotionController.TryGetReadingAtTime(currentTime);
+        }
+        public MotionController MotionController { get; private set; }
+        public MotionControllerReading CurrentReading { get; private set; }
+
+    }
+
+    [MixedRealityController(
+        SupportedControllerType.HPMotionController,
+        new[] { Toolkit.Utilities.Handedness.Left, Toolkit.Utilities.Handedness.Right },
+        flags: MixedRealityControllerConfigurationFlags.UseCustomInteractionMappings)]
```

Comment:
I don't think this needs this flag, since it doesn't use custom mappings from the profile. We'll want to explicitly define the mappings in this class by overriding `DefaultInteractions`, which I might be overlooking but I don't see the new mappings in here.

Reply:
Updated with the proper mappings (I think!)